### DISCUSSION TITLE
Aspect ratio

### DISCUSSION THREAD

**Discussion by FrnchFrgg** [dbuezas/lovelace-plotly-graph-card#83] [Q&A]
I use `layout-card` in CSS grid mode to have two columns in my weather lovelace view. When the window width is less than a certain value, the two columns collapse into a single one. The important part is that I set up the temperatures graph always spanning the full width, that is the total number of columns that are displayed.

As is, the plotly graph card uses a fixed height whatever the current width is. I would like it to set its height from a given aspect-ratio but did not succeed in doing so. I tried setting the `aspect-ratio` CSS property of the `ha-card` with `card-mod` but that only works while editing the card. Is it a bug with `card-mod`, some incompatibility between `card-mod` and the way `plotly` works ?

Anyway, it would be much better and less of a hack to have a dedicated setting to set aspect-ratio. Is there a possibility to do so already or should I file a feature request ?

**Comment by dbuezas** [comment]
Indeed not supported yet, so feature request it is.
It should be quick to implement, do you want to give it a try or should I just do it?

**Comment by dbuezas** [comment]
Pro tip: i only look at the issues page when I want to code something. If you need this, raise an issue :)

**Reply by netweaver1970** [reply]
I'm also interested in this, as I have few full-width graphs. And they look (too) skinny. They could use some bigger vertical size, so some aspect ratio setting would be nice.

**Reply by dbuezas** [reply]
In the meanwhile, try width, height (search in readme). Also try the pane view in home assistant
